src/app/keycloak-setup.ts

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2015-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+'use strict';
+
+import { CheBranding } from '../components/branding/che-branding';
+
+declare const Keycloak: Function;
+
+export interface IKeycloakAuth {
+  isPresent: boolean;
+  keycloak: any;
+  config: any;
+}
+
+export async function keycloakSetup(cheBranding: CheBranding): Promise<IKeycloakAuth> {
+  const keycloakAuth = {
+    isPresent: false,
+    keycloak: null,
+    config: null
+  };
+
+  let hasSSO = false;
+
+  // load keycloak settings
+  let keycloakSettings;
+  try {
+    keycloakSettings = await angular.element.get('/api/keycloak/settings');
+  } catch (e) {
+    const errorMessage = `Can't get Keycloak settings: ` + e.statusText;
+    console.warn(errorMessage);
+    throw new Error(errorMessage);
+  }
+
+  if (!keycloakSettings['che.keycloak.js_adapter_url']) {
+    return;
+  }
+  hasSSO = true;
+
+  keycloakAuth.config = buildKeycloakConfig(keycloakSettings);
+
+  // load keycloak adapter
+  // initialize keycloak
+  let keycloak;
+  try {
+    await loadKeycloakAdapter(keycloakSettings, cheBranding)
+    keycloak = await initKeycloak(keycloakAuth.config, keycloakSettings);
+
+    keycloakAuth.isPresent = true;
+    keycloakAuth.keycloak = keycloak;
+    /* tslint:disable */
+    window['_keycloak'] = keycloak;
```

Comment:
Why do we need this object ```window['_keycloak'] = keycloak;``` when we already have one ```keycloakAuth.keycloak = keycloak;```?

Reply:
I think we should use only one.

Reply:
good catch! it's not used after this refactoring, I'll remove it.

Reply:
done